calc/calc_mamba_flops.py

```diff
@@ -59,160 +66,187 @@ def config_parser():
+                        default=300e9,
                         help='Number of tokens you are training over')
     parser.add_argument("--no-checkpoint-activations", "-ca",
                         action='store_false',
                         help='Whether activation checkpointing is being used',
                         dest='checkpoint_activations')
-    parser.add_argument("--mamba_moe_layers", type = str, default = "")
+    parser.add_argument("--mamba-moe-layers",
+                        type=str,
+                        default="",
+                        help='Layer configuration string')
     return parser
 
-def compute_mamba2_flops(args):
-    d_inner = args.hidden_size * args.expand
-    Nheads = d_inner // args.mamba_headdim
-    mamba2_block_flops = 2 * (2 * d_inner + 2 * args.mamba_ngroups  * args.state_size + Nheads) * args.hidden_size * args.batch_size * args.sequence_length # in proj
-    mamba2_block_flops += 2 * args.batch_size * args.sequence_length * (d_inner + 2 * args.mamba_ngroups * args.state_size) * args.conv_dimension * args.d_inner# conv
-    mamba2_block_flops += 2 * args.batch_size * args.sequence_length * d_inner * args.state_size * args.d_inner # dtbx
-    mamba2_block_flops += 2 * args.batch_size * args.sequence_length * d_inner * args.state_size # ssm state rollover
-    mamba2_block_flops += 2 * args.batch_size * args.sequence_length * d_inner * args.state_size * args.d_model # c-> y 
-    mamba2_block_flops += args.batch_size * args.sequence_length * args.hidden_size # z gate output
-    return mamba2_block_flops
-
-def compute_mamba_flops(args):
+def compute_mamba1_flops(args, iter_factor):
     d_inner = args.hidden_size * args.expansion_factor
-    dt_rank = math.ceil(args.hidden_size / 16) if args.dt_rank == "auto" else args.dt_rank
+    dt_rank = math.ceil(args.hidden_size / 16) if args.dt_rank == "auto" else int(args.dt_rank)
+    
+    # SSM state computation
     ssm_flops = iter_factor * d_inner * args.tokens * (11 * args.state_size + 4 * dt_rank + 1)
+    # Input and output projections
     mamba_projectors_flops = iter_factor * args.tokens * 6 * d_inner * args.hidden_size 
+    # Convolution operations
     mamba_conv_flops = iter_factor * args.tokens * 2 * d_inner * args.conv_dimension
-    mamba_flops = ssm_flops + mamba_projectors_flops + mamba_conv_flops
-    return mamba_flops
+    
+    mamba1_flops = ssm_flops + mamba_projectors_flops + mamba_conv_flops
+    return mamba1_flops
+
+def compute_mamba2_flops(args):
+    d_inner = args.hidden_size * args.expansion_factor
+    Nheads = d_inner // args.mamba_headdim
+    
+    # Input projections
+    mamba2_block_flops = 2 * (2 * d_inner + 2 * args.mamba_ngroups * args.state_size + Nheads) * args.hidden_size * args.batch_size * args.sequence_length
+    # Convolution computations
+    mamba2_block_flops += 2 * args.batch_size * args.sequence_length * (d_inner + 2 * args.mamba_ngroups * args.state_size) * args.conv_dimension * d_inner
```

Comment:
```suggestion
    mamba2_block_flops += 2 * args.batch_size * args.sequence_length * (d_inner + 2 * args.mamba_ngroups * args.state_size) * args.conv_dimension + (d_inner + 2 * args.mamba_ngroups * args.state_size) * args.batch_size * args.sequence_length
```

drop one factor of `d_inner` and add contribution from the SiLU that follows the conv